filewatcher/filewatcher.go

```diff
@@ -97,27 +119,57 @@ func getMtime(path string) (time.Time, error) {
 
 func (r *Result) watcherLoop(
 	watcher *fsnotify.Watcher,
+	watched map[string]bool,
```

Comment:
nit: since the list is generated by `filepath.WalkDir`, there shouldn't be any concerns of duplications, so just a `[]string` should be enough.